### PR TITLE
New virtual profile to clone biggest screen

### PR DESCRIPTION
**Pull request by adrianlzt**
This option set the biggest monitor with it preferred configuration and
scale the others monitors to show it completely

### CODE REVIEW

File: autorandr.py [hunk 1]
```diff
@@ -715,6 +716,20 @@ def generate_virtual_profile(configuration, modes, profile_name):
                 shift += int(mode[shift_index])
             else:
                 configuration[output].options["off"] = None
+    elif profile_name == "clone":
+        biggest_width_resolution = sorted([output_modes[0] for output, output_modes in modes.items()], key=lambda x: x["width"], reverse=True)[0]
```
Comment (phillipberndt) [inline autorandr.py]: There are people using screens in vertical mode (9:16). How about sorting by `width*height` instead?

File: autorandr.py [hunk 2]
```diff
@@ -715,6 +716,20 @@ def generate_virtual_profile(configuration, modes, profile_name):
...
+        for output in configuration:
+            configuration[output].options = {}
+            if output in modes and configuration[output].edid:
+                mode = sorted(modes[output], key=lambda a: int(a["width"])*int(a["height"]) + (10**6 if a["preferred"] else 0))[-1]
```
Comment (phillipberndt) [inline autorandr.py]: Would it be a good idea to prefer modes that have the same aspect ratio as `biggest_width_resolution`? Or: Would it be possible to maintain the aspect ratio when scaling below, adding black borders around the view? (idk myself, but making the screen a little larger than required and panning everything to the center might work?!)

**Comment by adrianlzt** [comment]
Is not finished.
I have some problems with this implementation.

With two screens configured with horizontal vertical profile, if I execute this new clone profile:
```
Loading profile 'clone'
| Differences between the two profiles:
| [Output eDP1] Option --primary is not present in the new configuration
| [Output eDP1] Option --scale (`1.4055636896046853x1.40625' in the new configuration) is not present currently
| [Output HDMI1] Option --pos (= `1366x0') is `0x0' in the new configuration
| [Output HDMI1] Option --scale (`1.0x1.0' in the new configuration) is not present currently
\-
Going to run:
xrandr --output eDP1 --gamma 1.0:1.0:1.0 --mode 1366x768 --pos 0x0 --rate 60.00 --reflect normal --rotate normal --scale 1.4055636896046853x1.40625 --output HDMI1 --gamma 1.0:1.0:1.0 --mode 1920x1080 --pos 0x0 --rate 60.00 --reflect normal --rotate normal --scale 1.0x1.0
The configuration change did not go as expected:
| Differences between the two profiles:
| [Output eDP1] Option --transform (= `1.405563,0.000000,0.000000,0.000000,1.406250,0.000000,0.000000,0.000000,1.000000') is not present in the new configuration
| [Output eDP1] Option --primary is not present in the new configuration
| [Output eDP1] Option --scale (`1.4055636896046853x1.40625' in the new configuration) is not present currently
| [Output HDMI1] Option --scale (`1.0x1.0' in the new configuration) is not present currently
\-
```
Works as expected.

Then I execute the same command:
```
Loading profile 'clone'
| Differences between the two profiles:
| [Output HDMI1] Option --scale (`1.0x1.0' in the new configuration) is not present currently
| [Output eDP1] Option --transform (= `1.405563,0.000000,0.000000,0.000000,1.406250,0.000000,0.000000,0.000000,1.000000') is not present in the new configuration
| [Output eDP1] Option --primary is not present in the new configuration
| [Output eDP1] Option --scale (`1.4055636896046853x1.40625' in the new configuration) is not present currently
\-
Going to run:
xrandr --output eDP1 --transform none
xrandr --output eDP1 --gamma 1.0:1.0:1.0 --mode 1366x768 --pos 0x0 --rate 60.00 --reflect normal --rotate normal --scale 1.4055636896046853x1.40625 --transform 1,0,0,0,1,0,0,0,1 --output HDMI1 --gamma 1.0:1.0:1.0 --mode 1920x1080 --pos 0x0 --rate 60.00 --reflect normal --rotate normal --scale 1.0x1.0
The configuration change did not go as expected:
| Differences between the two profiles:
| [Output HDMI1] Option --scale (`1.0x1.0' in the new configuration) is not present currently
| [Output eDP1] Option --primary is not present in the new configuration
| [Output eDP1] Option --scale (`1.4055636896046853x1.40625' in the new configuration) is not present currently
\-
```
After this the big screen just uses a portion of the screen to clone the other smaller screen.

I saw the difference is:
```
--transform 1,0,0,0,1,0,0,0,1
```
But I don't know how to avoid setting that parameter.


I am also supposing that the biggest screen resolution is always the first one for each device. Not sure if that is true.

**Comment by phillipberndt** [comment]
> But I don't know how to avoid setting that parameter.

Have a look at the documentation of `--transform 1,0,0,0,1,0,0,0,1`. It' equivalent to scale 1x1, you can use `--transform` instead of `--scale`.

**Comment by phillipberndt** [comment]
Thanks for making the changes! I'll see that I'll manage to test this tomorrow. I don't have a suitable secondary screen currently 😔

**Comment by RasmusWL** [comment]
I just noticed this PR, I think it looks like a neat option to have :)

However, I'm not sure if the name `clone` is precise enough. Just from the name, I would guess it should be the same as `common`. Naming is always hard, but how about something along the lines of `clone-largest`? (you guys can probably come up with better namings, I just wanted to highlight this potential problem)

**Comment by adrianlzt** [comment]
Yeah, probably the name is not perfect.
@phillipberndt you decide

**Comment by phillipberndt** [comment]
Good point. `clone-largest` sounds good to me.

@adrianlzt Were you able to test this with monitors with different resolutions? I experience some issues: I forced one of my monitors to 1024x768 (by hacking the mode within in the for loop, mind to make of copy of the dictionary if this mode also is the biggest resolution, otherwise you'll change both) to see what happens. Biggest resolution is Full HD. autorandr tried to run

    xrandr --output HDMI-0 --gamma 1.0:1.0:1.0 --mode 1024x768 --pos 0x0 --rate 60.00 --reflect normal --rotate normal --transform 1.875,0,0,0,1.875,-180,0,0,1 

which seems to be correct. But this didn't work, and left the screen in an unusable state. My PC seems to have a problem in principle with scale factors greatly exceeding unity. This probably is my graphics driver, but it wouldn't make sense to merge this unless it's working for someone - does it do for you?

**Comment by adrianlzt** [comment]
Yeah, it works.
First one is the "common" profile and second one the "clone-largest".
common
clone-largest

**Comment by phillipberndt** [comment]
Great. Thanks again! I have added you to the authors list, let me know if you'd prefer to see a pseudonym there.

**Pull request merged**